presto-spi/src/test/java/io/prestosql/spi/testing/InterfaceTestUtils.java

```diff
@@ -64,11 +64,22 @@ private InterfaceTestUtils() {}
                     }));
 
             try {
+                if (!defines(forwardingInstance, actualMethod)) {
+                    continue;
+                }
+
                 actualMethod.invoke(forwardingInstance, actualArguments);
             }
             catch (Exception e) {
                 throw new RuntimeException(format("Invocation of %s has failed", actualMethod), e);
             }
         }
     }
+
+    private static boolean defines(Object forwardingInstance, Method actualMethod)
+            throws Exception
+    {
+        Class<?> forwardingInstanceClass = forwardingInstance.getClass();
```

Comment:
I think this would be easier to read with shorter variable names. Also, introduce a variable
```java
Class<?> forwardingClass = forwardingInstance.getClass();
Method forwardingMethod = forwardingClass.getMethod(method.getName().method.getParameterTypes());
return forwardingClass == forwardingMethod.getDeclaringClass();